plugins/woocommerce/includes/class-wc-discounts.php

```diff
@@ -969,8 +969,16 @@ protected function validate_coupon_allowed_emails( $coupon ) {
 			return true;
 		}
 
+		$check_emails = array();
 		$user         = wp_get_current_user();
-		$check_emails = array( $user->get_billing_email(), $user->get_email() );
+
+		try {
+			$customer     = new WC_Customer( $user->ID );
+			$check_emails = array( $customer->get_billing_email(), $customer->get_email() );
+		} catch ( Exception $e ) {
+			$wc_logger = wc_get_logger();
+			$wc_logger->debug( $e->getMessage() );
+		}
```

Comment:
Perhaps we can avoid using the WC_Customer object, and dealing with a possible Exception, and also make this on par with an existing validation on [class-wc-cart.php#L1663](https://github.com/woocommerce/woocommerce/blob/86bc37797707b822324c241761922b8bfa9717e8/plugins/woocommerce/includes/class-wc-cart.php#L1663) by using:

`$check_emails = array( $user->user_email );`

Then billing email would come from customer (via WC_Cart) or directly from WC_Order.

An alternate way to this would be resorting to wc()->customer, but I believe $user->user_email would be enough to cover the account email.